SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1428,6 +1430,29 @@ \section{Pipelines}
 
 \end{itemize}
 
+\section{The auto Keyword}
+\label{sec:TheAutoKeyword}
+
+Available since C++11, the \code{auto} keyword specifies that a variable's type
+is automatically deduced from its initializer.
+
+The \code{auto} keyword should be used to specify a type in the following cases:
+
+\begin{itemize}
+\item The type is duplicated on the left side of an assigment when it is mandated on the
+      right side, e.g. when there is an explicit cast or initializing with \code{new}.
+\item When obtaining container elements, when the element type is obvious from the type
+      of the container.
+\item When declaring iterators in a \code{for} loop.
+\end{itemize}
+
+All other cases should not use \code{auto}, but a semantic type name
+should be used that convey's meaning, as described in Section~\ref{sec:NamingConventions}
```

Comment:
`should be used that conveys meaning`? (i.e. without apostrophe: typo?)

Reply:
Good catch. Addressed.